### PR TITLE
Remove concept of User and UserManager from web3 providers

### PR DESCRIPTION
**Pull request by pixelmatrix**

### CODE REVIEW

File: src/providers/bitski-provider.ts [hunk 1]
```diff
@@ -182,7 +184,7 @@ export class BitskiProvider extends OAuthHttpProvider {
 
         const ethSendTransactionUrl = `${BITSKI_WEB_HOST}/eth-send-transaction`;
         const encodedPayload = btoa(JSON.stringify(payload));
-        const txnParams = `network=${this.networkName}&payload=${encodedPayload}&accessToken=${user.access_token}`;
+        const txnParams = `network=${this.networkName}&payload=${encodedPayload}&accessToken=${accessToken}`;
```
Comment (jtescher) [inline src/providers/bitski-provider.ts]: Can we switch this to `referrerAccessToken` to disambiguate in web UI?

**Comment by pat2man** [comment]
@pixelmatrix could we just ditch OAuthHTTPProvider and just use the fact that HTTPProvider can accept headers?

File: src/providers/bitski-provider.ts (hunk 1 above)
Comment (pixelmatrix) [reply src/providers/bitski-provider.ts]: Updated this

**Comment by pixelmatrix** [comment]
We discussed keeping OAuthHTTPProvider as a primitive HTTPProvider that simply passes in an auth header for now. That way we don't have to manage the array of headers separately as an auth token can change. BitskiProvider adds the functionality to require an auth token, require authorization, and defer sends until authenticated.

**Pull request merged**